go mod replacements breaks docker build

### Describe the bug
The lib path replacements in `go.mod` will break the docker build cause these libs are out of dockers context. 

### Reproduce
`docker build services/user-svc/`

### Logs
```
Sending build context to Docker daemon  54.27kB
Step 1/15 : FROM golang:1.19 AS builder
 ---> 8827cedaa309
Step 2/15 : WORKDIR /app
 ---> Using cache
 ---> d763ff2635d0
Step 3/15 : COPY . /app
 ---> Using cache
 ---> 13d2a73edae4
Step 4/15 : ARG VERSION
 ---> Using cache
 ---> cba1e0088c93
Step 5/15 : RUN CGO_ENABLED=0 GOOS=linux go build -ldflags="-X main.Version=${VERSION}" -a -o app .
 ---> Running in b06857608dae
go: downloading golang.org/x/crypto v0.0.0-20220722155217-630584e8d5aa
go: downloading github.com/google/uuid v1.3.0
go: downloading github.com/dapr/go-sdk v1.6.0
go: downloading github.com/rs/zerolog v1.28.0
go: downloading github.com/mattn/go-colorable v0.1.12
go: downloading github.com/mattn/go-isatty v0.0.14
go: downloading golang.org/x/sys v0.0.0-20220520151302-bc2c85ada10a
main.go:4:2: common@v0.0.0: replacement directory ../../libs/common does not exist
main.go:9:2: hwgorm@v0.0.0: replacement directory ../../libs/hwgorm does not exist
main.go:10:2: hwutil@v0.0.0: replacement directory ../../libs/hwutil does not exist
main.go:11:2: logging@v0.0.0: replacement directory ../../libs/logging does not exist
The command '/bin/sh -c CGO_ENABLED=0 GOOS=linux go build -ldflags="-X main.Version=${VERSION}" -a -o app .' returned a non-zero code: 1
```

### Additional Comments
We have to restructure the project so that docker can use that shared lib directory!

--- a/.build/service.Dockerfile
+++ b/.build/service.Dockerfile
@@ -1,16 +1,26 @@
+# builder builds a go service
+# the build args SERVICE and VERSION must be set!
 FROM golang:1.19 AS builder
 
-WORKDIR /app
-COPY . /app
-
+ARG SERVICE
 ARG VERSION
+
+WORKDIR /build/app
+
+COPY libs /libs
+
+COPY services/$SERVICE/go.* ./
+RUN go mod download
+
+COPY services/$SERVICE /build/app
+
 RUN CGO_ENABLED=0 GOOS=linux go build -ldflags="-X main.Version=${VERSION}" -a -o app .
 
 FROM alpine AS production
 
 WORKDIR /app
 
-COPY --from=builder /app/app .
+COPY --from=builder /build/app/app .
 RUN chmod +x app
 
 ENV PORT 80
@@ -21,4 +31,3 @@ ENV POSTGRES_PORT 5432
 EXPOSE 80
 
 CMD ["./app"]
-

--- a/.dockerignore
+++ b/.dockerignore
@@ -3,4 +3,3 @@ README.md
 Dockerfile
 docker-compose.yml
 .*
-

--- a/Dockerfile.service
+++ b/Dockerfile.service
@@ -1,3 +1,6 @@
+# you probably don't want to use this Dockerfile directly,
+# use `make <svc>` instead
+
 # builder builds a go service
 # the build args SERVICE and VERSION must be set!
 FROM golang:1.19 AS builder

--- a/Makefile
+++ b/Makefile
@@ -5,7 +5,7 @@ SERVICES = user-svc emergency-room-svc
 
 .PHONY: $(SERVICES)
 $(SERVICES):
-	docker build -f ${DOCKERFILE_SERVICES} --build-arg=VERSION=development --build-arg=SERVICE=$@ -t helpwave/$@ .
+	docker build -f ${DOCKERFILE_SERVICES} --build-arg=VERSION=${VERSION} --build-arg=SERVICE=$@ -t helpwave/$@ .
 
 .PHONY: clean
 clean:

--- a/Makefile
+++ b/Makefile
@@ -1,0 +1,13 @@
+VERSION ?= $(shell git log --format="%h" -n 1)
+DOCKERFILE_SERVICES = .build/service.Dockerfile
+
+SERVICES = user-svc emergency-room-svc
+
+.PHONY: $(SERVICES)
+$(SERVICES):
+	docker build -f ${DOCKERFILE_SERVICES} --build-arg=VERSION=development --build-arg=SERVICE=$@ -t helpwave/$@ .
+
+.PHONY: clean
+clean:
+	docker rmi $$(docker images --format '{{.Repository}}:{{.Tag}}' | grep "^helpwave/")
+	yes | docker image prune

--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,5 @@
 VERSION ?= $(shell git log --format="%h" -n 1)
-DOCKERFILE_SERVICES = .build/service.Dockerfile
+DOCKERFILE_SERVICES = Dockerfile.service
 
 SERVICES = user-svc emergency-room-svc
 

--- a/README.md
+++ b/README.md
@@ -1,3 +1,22 @@
 # services
 
 helpwave's microservices
+
+## Docker images
+
+You can build a docker image for a service using `make`:
+```bash
+make <servicename>
+```
+ e.g.:
+```bash
+ make user-svc
+```
+
+Each service should have a Makefile, that includes the root's Makefile.
+In the context of a service directory you should be able to use
+```bash
+ make this
+```
+to build the corresponding docker image as well.
+

--- a/services/emergency-room-svc/Makefile
+++ b/services/emergency-room-svc/Makefile
@@ -1,0 +1,7 @@
+# include top-level Makefile
+%:
+	$(MAKE) -C ../../. $@
+
+.PHONY: this
+this:
+	$(MAKE) -C ../../. $$(basename "$$PWD")

--- a/services/user-svc/Makefile
+++ b/services/user-svc/Makefile
@@ -1,0 +1,7 @@
+# include top-level Makefile
+%:
+	$(MAKE) -C ../../. $@
+
+.PHONY: this
+this:
+	$(MAKE) -C ../../. $$(basename "$$PWD")

